examples/Constraints_Continuous/interpoint.py

```diff
@@ -0,0 +1,264 @@
+# Second, it shows how to also include a **solvent budget** constraint for controlling
+# the total solvent consumption across experiments for cost efficiency.
+
+# This example demonstrates how to use interpoint constraints and intrapoint constraints.
+# An intrapoint constraint, often simply referred to as a constraint, applies to each individual
+# experiment, ensuring that certain conditions are met within that single point.
+# In contrast, an interpoint constraint applies across a batch of experiments,
+# enforcing conditions that relate to the collective set of points rather than
+# individual ones. These constraints are particularly useful when resources or conditions must be
+# managed at a batch level and they allow us to:
+# * Ensure total resource consumption meets exact requirements
+# * Maintain chemical balances across multiple experiments
+# * Optimize the collective use of expensive materials
+
+# For more details on interpoint constraints, see the {ref}`user guide on constraints
+# <userguide/constraints:ContinuousLinearConstraint>`.
+
+# ## Imports and Settings
+
+import os
+
+import pandas as pd
+from matplotlib import pyplot as plt
+
+from baybe import Campaign
+from baybe.constraints import ContinuousLinearConstraint
+from baybe.parameters import NumericalContinuousParameter
+from baybe.recommenders import BotorchRecommender
+from baybe.searchspace import SearchSpace
+from baybe.targets import NumericalTarget
+from baybe.utils.dataframe import add_fake_measurements
+from baybe.utils.random import set_random_seed
+
+SMOKE_TEST = "SMOKE_TEST" in os.environ
+BATCH_SIZE = 3
+N_ITERATIONS = 4 if SMOKE_TEST else 15
+TOLERANCE = 0.01
+
+
+set_random_seed(1337)
+
+# ## Defining the Chemical Optimization Problem
+
+# We'll optimize a synthetic chemical reaction with the following experimental parameters:
+# - **Solvent Volume** (10-30 mL per experiment): The amount of solvent used
+# - **Reactant A Concentration** (0.1-2.0 g/L): Primary reactant concentration
+# - **Catalyst Loading** (1-10 mol%): Catalyst amount as percentage of limiting reagent
+# - **Temperature** (60-120 °C): Reaction temperature
+# Note that these ranges are chosen arbitrary and do not represent a specific real-world reaction.
+
+parameters = [
+    NumericalContinuousParameter(
+        name="Solvent_Volume", bounds=(10.0, 30.0), metadata={"unit": "mL"}
+    ),
+    NumericalContinuousParameter(
+        name="Reactant_A_Conc", bounds=(0.1, 2.0), metadata={"unit": "g/L"}
+    ),
+    NumericalContinuousParameter(
+        name="Catalyst_Loading", bounds=(1.0, 20.0), metadata={"unit": "mol%"}
+    ),
+    NumericalContinuousParameter(
+        name="Temperature", bounds=(60.0, 120.0), metadata={"unit": "°C"}
+    ),
+]
+
+# ## Constraint Definition
+
+# We define both intrapoint and interpoint constraints to demonstrate the difference:
+#
+# **Intrapoint constraint** (applied to each individual experiment):
```

Comment:
The rendered version doesn't really look nice here:
* We have both numbers and bullets
* Bold font appears both in headings as well as at beginning of lines (and is missing in the first line)
* Indentation somehow just looks weird.

Can we perhaps find different representation of the same information that looks more pleasing? I don't know, perhaps a table could work? Or something else?